src/jit/hwintrinsicxarch.cpp

```diff
@@ -736,6 +736,8 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
     var_types baseType = TYP_UNKNOWN;
     var_types retType  = TYP_UNKNOWN;
 
+    assert((simdSize == 16) || (simdSize == 0));
```

Comment:
This could use a comment; I think that whenever we do the "simdSize == 0" check it deserves a comment, e.g. "A simdSize of zero means that it is not a SIMD type" because it could be confusing.

Reply:
The SSE code added:
> // The Prefetch and StoreFence intrinsics don't take any SIMD operands
> // and have a simdSize of 0